core/src/main/java/com/kin/ecosystem/core/accountmanager/AccountManagerImpl.java

```diff
@@ -182,19 +238,37 @@ public void onResponse(Boolean response) {
 				if (blockchainSource.updateActiveAccount(accountIndex)) {
 					callback.onResponse(response);
 				} else {
+					deleteRestoredAccount(accountIndex);
 					callback.onFailure(ErrorUtil.createAccountCannotLoadedException(accountIndex));
 				}
 			}
 
 			@Override
 			public void onFailure(KinEcosystemException exception) {
-				//switch to the new KinAccount
+				deleteRestoredAccount(accountIndex);
 				callback.onFailure(exception);
 				Logger.log(new Log().withTag(TAG).put("switchAccount", "ended with failure"));
 			}
 		});
 	}
 
+	/**
+	 * This method should be called in case the restore process is failed somewhere in the way. Because currently when
+	 * restoring an account the restored account is added to kinClient(to the end of the account list if it wasn't there
+	 * before) even before the restore process is finished successfully. That is why if the restore is failed we need to
+	 * delete the restored account from the end of the account list(if it was added), if it was already there then we
+	 * don't care about it because the active one will be the same as before.
+	 */
+	private void deleteRestoredAccount(int accountIndex) {
+		try {
+			Logger.log(new Log().withTag(TAG).put("deleteRestoredAccount", "account index = " + accountIndex));
+			blockchainSource.deleteAccount(accountIndex);
+		} catch (DeleteAccountException e) {
+			e.printStackTrace();
```

Comment:
can you send to the BI events logger also? the event is `GeneralEcosystemSdkError`